tests/test_eigendistortion.py

```diff
@@ -104,69 +103,44 @@ def test_method_power(self, color):
 
         assert ed.synthesized_signal.shape[-3:] == (n_chans, LARGE_DIM, LARGE_DIM)
 
-    @pytest.mark.parametrize("e_vecs", [[0, 1, -2, -1], []])
-    @pytest.mark.parametrize("color", [False, True])
-    def test_method_lanczos(self, e_vecs, color):
-        # return first and last two eigenvectors
-        n_steps = 5
-        n_chans = 3 if color else 1
-
-        # run once with e_vecs specified
-        ed = get_synthesis_object(im_dim=LARGE_DIM, color=color)
-
-        if n_steps < len(e_vecs) * 2:
-            with pytest.warns(RuntimeWarning) as not_enough_iter_warning:
-                ed.synthesize(method='lanczos', n_steps=n_steps, e_vecs=e_vecs)
-        else:
-            with pytest.warns(UserWarning) as lanczos_experimental_warning:
-                ed.synthesize(method='lanczos', n_steps=n_steps, e_vecs=e_vecs)
-
-        if len(e_vecs) > 0:
-            assert len(ed.synthesized_eigenvalues) == len(e_vecs)
-        else:
-            assert len(ed.synthesized_eigenvalues) == n_steps
-
-        assert len(ed.synthesized_signal) == len(e_vecs)
-        assert len(ed.synthesized_eigenindex) == len(e_vecs)
-
-        if len(e_vecs) > 0:
-            assert ed.synthesized_signal.shape[-3:] == (n_chans, LARGE_DIM, LARGE_DIM)
-
-    def test_lanczos_accuracy(self):
-        n = 30
-        e_vals = (torch.randn(n**2)**2).sort(descending=True)[0]
-        eigen_test_matrix = torch.diag(e_vals)
+    def test_orthog_iter(self):
+        n, k = 30, 10
+        n_chans = 1  # TODO color
         ed = get_synthesis_object(im_dim=n)
-        with pytest.warns(UserWarning) as lanczos_experimental_warning:
-            ed.synthesize(method='lanczos', n_steps=eigen_test_matrix.shape[-1], debug_A=eigen_test_matrix)
+        ed.synthesize(k=k, method='power', max_steps=10)
 
-        assert (e_vals[0]-ed.synthesized_eigenvalues[0]) < 1e-2
+        assert ed.synthesized_signal.shape == (k*2, n_chans, n, n)
+        assert ed.synthesized_eigenindex.allclose(torch.cat((torch.arange(k), torch.arange(n**2 - k, n**2))))
+        assert len(ed.synthesized_eigenvalues) == 2*k
 
-    def test_method_equivalence(self):
+    def test_method_svd(self):
+        n, k = 30, 10
```

Comment:
Should we test finding different number of eigenvectors (different values of `k`)? Is it guaranteed that the first eigenvector will always be the same (for different values of `k`) or should that be explicitly checked?

Reply:
I think diff values of ``k`` isn't necessary b/c the top should be equiv (within numerical precision) each time.